docs/source/reference/deprecation.rst

```diff
@@ -0,0 +1,327 @@
+--------
+This feature will be removed with respect to this schedule:
+
+* Deprecation warnings will be issued in version 0.44.0
+* Support will be removed in version 0.45.0
+
+Recommendations
+---------------
+Projects that need/rely on the deprecated behaviour should pin their dependency
+on Numba to a version prior to removal of this feature and consider migrating to
+using the ``numba.cuda`` API.
+
+
+Deprecation of ``NUMBAPRO`` environment variables
+=================================================
+As a piece of legacy behaviour inherited from the time of ``NumbaPro`` it is
+possible to specify various CUDA drivers, libraries and configuration to use
+with Numba via the following environment variables:
+
+* ``NUMBAPRO_NVVM``
+* ``NUMBAPRO_CUDALIB``
+* ``NUMBAPRO_LIBDEVICE``
+* ``NUMBAPRO_CUDA_LOG_SIZE``
+* ``NUMBAPRO_VERBOSE_CU_JIT_LOG``
+
+These are being deprecated with view of removal in the near future.
+
+Reason for deprecation
+----------------------
+First, ``NUMBAPRO`` as a name causes confusion for users, especially as the
+product ``NumbaPro`` was retired since its technology was predominantly made
+Open Source, with Numba inheriting a lot of this. Second, the environment
+variables are somewhat confusing and in some cases setting one will influence
+the impact of another, none of which is documented.
+
+Schedule
+--------
+This feature will be removed with respect to this schedule:
+
+* Deprecation warnings will be issued in version 0.44.0
+* Support will be removed in version 0.45.0
+
+Recommendations
+---------------
+Projects that need/rely on the deprecated behaviour should pin their dependency
+on Numba to a version prior to removal of this behaviour, or consider following
+replacement instructions that will be issued outlining how to adjust to the
+change.
+
+Expected Replacement
+--------------------
+Numba will gain a new set of CUDA related environment variables to replace the
+``NUMBAPRO`` environment variables. This change will be part of permitting
+discovery of system level CUDA drivers and libraries, as well as making it easy
+to point Numba to a particular CUDA SDK instance.
+
+
+Deprecation of ``numba.autojit``
+================================
+The decorator ``numba.autojit`` was inherited from the time of ``NumbaPro`` and
+has not been recommended for use for some time.
+
+Reason for deprecation
+----------------------
+It is functionally no different to ``numba.jit`` and in fact just calls out to
+this function. It has not been in active use or recommended for use for a
+significant period of time.
+
+Schedule
+--------
+This feature will be removed with respect to this schedule:
+
+* Deprecation warnings will be issued in version 0.44.0
+* Support will be removed in version 0.47.0
+
+Recommendations
+---------------
+Projects that need/rely on the deprecated behaviour should pin their dependency
+on Numba to a version prior to removal of this behaviour, or consider following
+the recommendations below.
```

Comment:
I think this recommendation was accidentally duplicated from a previous entry, since the next recommendation section seems like the correct one.

Reply:
Thanks, this is inconsistent, I've fixed up this paragraph now.